### PR TITLE
web: refactor build header

### PR DESCRIPTION
**Pull request by pivotal-jamie-klassen**
we started working on this to pursue #4235, but since switching priorities we want to avoid losing this work so let's just merge the refactoring part.

### CODE REVIEW

File: web/elm/src/Build/Output/Output.elm [hunk 1]
```diff
@@ -103,115 +97,102 @@ planAndResourcesFetched buildId ( plan, resources ) model =
...
 handleEnvelopes :
     List BuildEventEnvelope
     -> OutputModel
-    -> ( OutputModel, List Effect, OutMsg )
+    -> ( OutputModel, List Effect )
 handleEnvelopes envelopes model =
     envelopes
         |> List.reverse
-        |> List.foldr handleEnvelope ( model, [], OutNoop )
+        |> List.foldr handleEnvelope ( model, [] )
```
Comment (kcmannem) [inline web/elm/src/Build/Output/Output.elm]: cool function

**Comment by kcmannem** [comment]
yo this is dope, I can read the code but I don't have experience to think more then just the changes look good at the surface. I learned a lot though. Thanks for including me, feel free to do it more often.

**Comment by vito** [comment]
While accepting this I noticed we currently show a 'loading' spinner on a job that has no builds:



Opened this as a separate issue since it's probably unrelated to this refactor: https://github.com/concourse/concourse/issues/4493

**Comment by pivotal-jamie-klassen** [comment]
@vito definitely not from this refactor -- that spinner has been there for years, and only became visible after the sidebar was introduced.

**Comment by vito** [comment]
Noticed some weirdness: if I'm viewing a running build and switch quickly between a finished build and the running build, the header kind of...fuses? with the running build's header:



Rolling back the refactor, it looks like this *wasn't* a problem before:

**Comment by pivotal-jamie-klassen** [comment]
@vito good catch, that's a real bug. I realized that #4487 doesn't actually depend on these changes, so let's not block it (or v5.6.0) on this one.

**Comment by pivotal-jamie-klassen** [comment]
@vito fixed it... although i am still feeling that this component could use more tests.

**Comment by jomsie** [comment]
i figure this'll finish this particular refactor, next we'll be using the algorithm-v3 branch to continue with #4235 (hopefully once that branch gets rebased from master)

**Pull request merged**